functions/Invoke-DbaLogShippingRecovery.ps1

```diff
@@ -0,0 +1,430 @@
+        
+        # Check if they commandlet is eecuted from a pipeline
+        if ($PSCmdlet.MyInvocation.ExpectingInput) {
+            $db = $_
+            
+            $query += "WHERE lsd.secondary_database = '$db' "
+        }
+        else {
+            # If just one database is set in the database parameter
+            if($Database -ne $null) {
+                $db = $Database[0]
+
+                $query += "WHERE lsd.secondary_database = '$db'"
+            }
+        }
+        
+        # Add an extra line to easier read the query when in verbose mode
+        $query +="
+----------------------------------------------------------------"
+        #endregion Query setup
+
+        # Retrieve the log shipping information from the secondary instance
+        try {
+            Write-Message -Message "Retrieving log shipping information from the secondary instance" -Level 5 -Silent $Silent
+            $logshipping_details = Invoke-Sqlcmd2 -ServerInstance $SqlServer -Database 'msdb' -Query $query 
+        }
+        catch {
+            Stop-Function -Message ("Error retrieving the log shipping details: " + $_.Exception.Message) -Silent $Silent -InnerErrorRecord $_ -Target $SqlServer
+            
+            return
+        }
+
+        #region CIM Session creation
+        Write-Message -Level Verbose -Message "Creating CimSession on $ServerName over WSMan"
+
+        # Check if the CIM session needs to be created using a credential
+        if (!$Credential){
+            $CIMSession = New-CimSession -ServerName $ServerName -ErrorAction SilentlyContinue
+        }
+        else{
+            $CIMSession = New-CimSession -ServerName $ServerName -ErrorAction SilentlyContinue -Credential $SqlCredential
+        }
+        
+        # Create a CIM session if it's not yet created
+        if ($CIMSession.id -eq $null){
+            Write-Message -Level Verbose -Message "Creating CimSession on $ServerName over WSMan failed. Creating CimSession on $ServerName over DCom"
+            
+            if (!$Credential){
+                $CIMSession = New-CimSession -ServerName $ServerName -SessionOption $SessionOption -ErrorAction SilentlyContinue -Credential $SqlCredential
+            }
+            else{
+                $CIMSession = New-CimSession -ServerName $ServerName -SessionOption $SessionOption -ErrorAction SilentlyContinue
+            }
+        }
+        
+        # Check if the CIM session was created successfully
+        if ($CIMSession.id -eq $null){
+            Stop-Function -Message "Can't create CimSession on $ServerName"
+        }
+        
+        # Checking the status of the SQL Server Agent service
+        Write-Message -Message ("Retrieving the status of the SQL Server Agent" + $ls.secondary_database + "'") -Level 2 -Silent $Silent
+
+        # Get the agent service from the CIM session
+        try{
+            if($InstanceName -eq ''){
```

Comment:
I'm going to change $instancename to 'MSSQLSERVER' for defaults. Otherwise, the Cim command is too general and can get multiple agents for all the instances.